// @vuepress/bundler-webpack/lib/dev/createDevServerConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDevServerConfig = void 0;
const path_1 = require("path");
const utils_1 = require("@vuepress/utils");
const resolvePort_1 = require("./resolvePort");
const trailingSlashMiddleware_1 = require("./trailingSlashMiddleware");
const createDevServerConfig = async (app, options) => ({
    allowedHosts: 'all',
    compress: true,
    devMiddleware: {
        publicPath: app.options.base,
        writeToDisk: false,
        stats: app.env.isDebug ? 'normal' : 'errors-warnings',
    },
    headers: {
        'access-control-allow-origin': '*',
    },
    historyApiFallback: {
        disableDotRule: true,
        rewrites: [{ from: /./, to: utils_1.path.join(app.options.base, 'index.html') }],
    },
    host: app.options.host,
    hot: true,
    setupMiddlewares: (middlewares, devServer) => {
        var _a, _b, _c;
        (_a = devServer.app) === null || _a === void 0 ? void 0 : _a.use(trailingSlashMiddleware_1.trailingSlashMiddleware);
        return ((_c = (_b = options.devServerSetupMiddlewares) === null || _b === void 0 ? void 0 : _b.call(options, middlewares, devServer)) !== null && _c !== void 0 ? _c : middlewares);
    },
    open: app.options.open,
    port: await (0, resolvePort_1.resolvePort)(app.options.port),
    static: {
        // `static.directory` will fail on Windows if we do not replace / with \
        directory: app.dir.public().replace('/', path_1.sep),
        publicPath: app.options.base,
        watch: {
            ignoreInitial: true,
            ignored: [
                // Do not watch node_modules
                'node_modules',
            ],
        },
    },
});
exports.createDevServerConfig = createDevServerConfig;
