// @vuepress/client/dist/index.js
import {
  ClientOnly,
  Content,
  Vuepress,
  pageData,
  pageDataEmpty,
  pageFrontmatterSymbol,
  pageHeadSymbol,
  pageHeadTitleSymbol,
  pageLangSymbol,
  pagesData,
  resolvers,
  routeLocaleSymbol,
  siteData,
  siteLocaleDataSymbol,
  updateHeadSymbol,
  usePageData,
  usePageFrontmatter,
  usePageHead,
  usePageHeadTitle,
  usePageLang,
  usePagesData,
  useRouteLocale,
  useSiteData,
  useSiteLocaleData,
  useUpdateHead,
  withBase
} from "./chunk-RTLFYRTT.js";

// src/index.ts
import { RouterLink, useRoute, useRouter } from "vue-router";

// src/helpers/defineClientAppEnhance.ts
var defineClientAppEnhance = (clientAppEnhance) => clientAppEnhance;

// src/helpers/defineClientAppSetup.ts
var defineClientAppSetup = (clientAppSetup) => clientAppSetup;
export {
  ClientOnly,
  Content,
  RouterLink,
  Vuepress,
  defineClientAppEnhance,
  defineClientAppSetup,
  pageData,
  pageDataEmpty,
  pageFrontmatterSymbol,
  pageHeadSymbol,
  pageHeadTitleSymbol,
  pageLangSymbol,
  pagesData,
  resolvers,
  routeLocaleSymbol,
  siteData,
  siteLocaleDataSymbol,
  updateHeadSymbol,
  usePageData,
  usePageFrontmatter,
  usePageHead,
  usePageHeadTitle,
  usePageLang,
  usePagesData,
  useRoute,
  useRouteLocale,
  useRouter,
  useSiteData,
  useSiteLocaleData,
  useUpdateHead,
  withBase
};
