// @vuepress/bundler-vite/lib/dev/createDev.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDev = void 0;
const utils_1 = require("@vuepress/utils");
const vite_1 = require("vite");
const plugins_1 = require("../plugins");
const createDev = (options) => async (app) => {
    var _a;
    const viteConfig = (0, vite_1.mergeConfig)({
        configFile: false,
        plugins: (0, plugins_1.createPlugins)({
            app,
            options,
            isServer: false,
            isBuild: false,
        }),
        // `clearScreen` won't take effect in `config` hook of plugin API
        clearScreen: false,
    }, (_a = options.viteOptions) !== null && _a !== void 0 ? _a : {});
    const server = await (0, vite_1.createServer)(viteConfig);
    await server.listen();
    server.config.logger.info(utils_1.chalk.cyan(`\n  vite v${require('vite/package.json').version}`) +
        utils_1.chalk.green(` dev server running at:\n`), {
        clear: !server.config.logger.hasWarned,
    });
    server.printUrls();
    return server.close.bind(server);
};
exports.createDev = createDev;
